jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/processor/AlbumByGenreUpnpProcessor.java

```diff
@@ -60,100 +52,17 @@ public ProcId getProcId() {
-            if (parent != null) {
-                container.setParentID(String.valueOf(parent.getId()));
-            }
-        }
-        return container;
-    }
-
-    private Container createContainer(MediaFile item, String index) {
-        GenreContainer container = new GenreContainer();
-        container.setParentID(getProcId().getValue());
-        container.setId(getProcId().getValue() + ProcId.CID_SEPA + index);
-        String comment = item.getComment();
-        container
-                .setTitle(util.isGenreCountAvailable() ? item.getName().concat(SPACE).concat(comment) : item.getName());
-        container.setChildCount(isEmpty(comment) ? 0 : Integer.parseInt(comment));
-        return container;
-    }
-
-    @Override
-    public int getDirectChildrenCount() {
-        return searchService.getGenresCount(true);
-    }
-
-    @Override
-    public List<MediaFile> getDirectChildren(long offset, long maxResults) {
-        return searchService.getGenres(true, offset, maxResults).stream().map(this::toMediaFile).toList();
-    }
-
-    private MediaFile toMediaFile(Genre g) {
-        MediaFile m = new MediaFile();
-        m.setId(-1);
-        m.setTitle(g.getName());
-        if (0 != g.getAlbumCount()) {
-            m.setComment(Integer.toString(g.getAlbumCount()));
-        }
-        m.setGenre(g.getName());
-        return m;
-    }
-
-    @Override
-    public MediaFile getDirectChild(String id) {
-        int index = Integer.parseInt(id);
-        List<Genre> genres = searchService.getGenres(true);
-        if (genres.size() > index) {
-            return toMediaFile(genres.get(index));
-        }
-        return null;
-    }
-
-    @Override
-    public int getChildSizeOf(MediaFile item) {
-        return searchService.getAlbumsByGenres(item.getName(), 0, Integer.MAX_VALUE, util.getGuestFolders()).size();
+    public List<MediaFile> getChildren(Genre item, long offset, long maxResults) {
+        return searchService.getAlbumsByGenres(item.getName(), (int) offset, (int) maxResults, util.getGuestFolders());
     }
 
     @Override
-    public List<MediaFile> getChildren(MediaFile item, long offset, long maxResults) {
-        if (-1 == item.getId()) {
-            return searchService.getAlbumsByGenres(item.getGenre(), (int) offset, (int) maxResults,
-                    util.getGuestFolders());
-        }
-        return mediaFileService.getSongsForAlbum(offset, maxResults, item);
+    public int getChildSizeOf(Genre genre) {
+        return genre.getAlbumCount();
     }
 
     @Override
-    public void addChild(DIDLContent didl, MediaFile song) {
-        if (isEmpty(song.getMediaType())) {
-            didl.addItem(factory.toMusicTrack(song));
-        } else {
-            didl.addContainer(createContainer(song));
-        }
+    public void addChild(DIDLContent didl, MediaFile album) {
+        didl.addContainer(factory.toAlbum(album, mediaFileService.getChildSizeOf(album)));
```

Comment:
The name is confusing. 

> DIDLContent didl

In this case `parent` is better. Fix not just here but overall.

Reply:
Done.